test/e2e/tests/test_permission.py

```diff
@@ -0,0 +1,155 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+# 	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+"""Integration tests for the Permission API.
+"""
+
+import pytest
+import time
+import json
+import logging
+import boto3
+
+from acktest.resources import random_suffix_name
+from acktest.k8s import resource as k8s
+from e2e import service_marker, CRD_GROUP, CRD_VERSION, load_ram_resource
+from e2e.replacement_values import REPLACEMENT_VALUES
+from e2e.bootstrap_resources import get_bootstrap_resources
+from e2e import ram_permission
+
+RESOURCE_PLURAL = "permissions"
+
+CREATE_WAIT_AFTER_SECONDS = 5
+MODIFY_WAIT_AFTER_SECONDS = 31
+DELETE_WAIT_AFTER_SECONDS = 20
+
+@pytest.fixture(scope="module")
+def permission():
+	resource_name = random_suffix_name("permission", 24)
+
+	resources = get_bootstrap_resources()
+	logging.debug(resources)
+
+	replacements = REPLACEMENT_VALUES.copy()
+	replacements["PERMISSION_NAME"] = resource_name
+
+	# Load Permission CR
+	resource_data = load_ram_resource(
+		"ram_permission",
+		additional_replacements=replacements,
+	)
+	logging.debug(resource_data)
+
+	# Create k8s resource
+	ref = k8s.CustomResourceReference(
+		CRD_GROUP, CRD_VERSION, RESOURCE_PLURAL,
+		resource_name, namespace="default",
+	)
+	k8s.create_custom_resource(ref, resource_data)
+	cr = k8s.wait_resource_consumed_by_controller(ref)
+	time.sleep(CREATE_WAIT_AFTER_SECONDS)
+	cr = k8s.get_resource(ref)
+	yield cr, ref
+
+	# Delete k8s resource
+	_, deleted = k8s.delete_custom_resource(
+		ref,
+		period_length=DELETE_WAIT_AFTER_SECONDS,
+	)
+	assert deleted
+
+@service_marker
+@pytest.mark.canary
+class TestPermission:
+	def test_crud(self, permission):
```

Comment:
can we also have a test for spec.tag updates?

Reply:
There isn't a way to update tags for Permissions

Reply:
We don't have a way to update Permission tags